framework/include/utils/MooseTypes.h

```diff
@@ -166,6 +166,8 @@ enum KernelType
 {
   KT_TIME = 0,
   KT_NONTIME = 1,
+  KT_NONEIGEN = 2,
+  KT_EIGEN  = 3,
```

Comment:
I followed  `time kernels` to add constants. I will remove constants. I have been trying my best to avoid changing the exiting code.